client-apis/encrypt.md

```diff
@@ -83,26 +95,62 @@ A bound on the length of the [plaintext](#plaintext) to encrypt.
 
 This behavior MUST output the following if the behavior is successful:
 
-- [message](../data-format/message.md)
+- [encrypted message](#encrypted-message)
+
+### Encrypted Message
+
+An encrypted form of the input [plaintext](#plaintext),
+encrypted according to the [operation specified below](#operation).
+This MUST be a sequence of bytes
+and conform to the [message format specification](../data-format/message.md).
+
+This operation MAY stream the encrypted message to the caller.
+
+- The operation MAY release the output message sequentially and temporally to the caller.
+- The operation MUST be able to indicate an end to the released sequential output.
+
+## Operation
 
-To construct the outputs, some fields MUST be constructed using information obtained
+The Encrypt operation is divided into several distinct parts:
+
+- [Get the encryption materials](#get-the-encryption-materials)
+- [Construct the header](#construct-the-header)
+- [Construct the body](#construc-the-body)
+- [(Optional) Construct the signature](#construct-the-signature)
+
+This operation MUST perform these steps in order.
+These steps calculate and serialize the components of the output [encrypted message](#encrypted-message).
+Any data that is not specified within the [message format](../data-format/message.md)
+MUST NOT be added to the output message.
+
+The bytes of the output encrypted message MAY be released to the caller as soon as they are serialized.
```

Comment:
I don't think so...?
The encrypt operation MUST know where the end is.
Especially for framed messages.
I suppose it could be up to the implementation,
but this seems like something that is easy to get wrong.

e.g. Frame Length is 4K, and I send 100 byte chunks
and then send EOF after 10 chunks (1K).
The header can be released, but
if the body header has been sent
it is probably wrong.

Also, the plaintext length bound
is just an estimate,
so I don't think we can rely on it
as a single to send the final frame.

Reply:
I'm not sure I understand your example. If the operation has successfully serialized a frame, it should be able to release that frame.

Reply:
Note: Update to SHOULD.

We do not state what it means to be "serialized", so there are parts where this wording is confusing.

Reply:
Update: staying with MAY because I'm unsure how to phrase this while not implying that the operation is the necessarily the actor for releasing the bytes. I think we are still ok as far as stating earlier that you SHOULD NOT extend a streaming API if you need to buffer the whole message.